### ISSUE TITLE
Dummy actor creation fails in PF2e

### ISSUE THREAD

**Issue by danheuck** [shemetz/roll-from-compendium#21]
It seems like there was a change in PF2e which is causing the module to fail when it attempts to create the dummy actor:

![image](https://github.com/shemetz/roll-from-compendium/assets/2694437/1d9c9a38-0d07-4760-bea0-8f6913a950c7)

The issue seems to be at [quick-send-to-chat.js:270](https://github.com/shemetz/roll-from-compendium/blob/ee9ea9afe953cdb7a8293ceef2bfc314f17177f7/scripts/quick-send-to-chat.js#L270C59-L270C59)

```
    type: types[0],  // e.g. 'character' in dnd5e and pf2e
    types: types[0],
```

Presumably `types[0]` used to be `character` but it's now `base` which seems to be causing the constructor to fail:

![image](https://github.com/shemetz/roll-from-compendium/assets/2694437/5309107d-f54d-42c6-9c8e-c0af4743d5a7)

If I modify those two lines to use types[1], it seems to work (the message gets posted to chat), but there's still an error in the console for spell casts (and maybe other things, I haven't checked extensively):

![image](https://github.com/shemetz/roll-from-compendium/assets/2694437/13df9c04-4b19-4797-a889-385943b95326)

This seems to be due to the PF2e system doing some checks where it assumes a token exists to cast the spell, but it works if you aren't looking at the log, so I wouldn't call it critical:

```
function getCondition(token, target, isSpell) {
  const checkingAttacker = target === null;
  const currentActor = checkingAttacker ? token?.actor : target?.actor;
  const conditionMap = checkingAttacker ? { ...actorConditionMap } : targetConditionMap;
  const attackerBlinded = !!token.actor?.items?.find(i => i.slug === "blinded");
  const attackerDazzled = !!token.actor?.items?.find(i => i.slug === "dazzled");
  const attackerHasBlindFight = !!token.actor?.items?.find((i) => i.slug === "blind-fight");
  const attackerHasLiminalFetchling = !!token.actor?.items?.find((i) => i.slug === "liminal-fetchling");
  ...
```

**Comment by shemetz** [comment]
Thanks for the detailed report!

I fixed the main bug here now, v1.7.1.

I can't reproduce the console errors, though.  Can you try doing it again after updating?  I wonder if it's a weird race condition.

**Comment by danheuck** [comment]
Sorry, looks like the console errors were due to another module I had accidentally left enabled during testing, [PF2e Flat Check](https://foundryvtt.com/packages/pf2-flat-check); not an issue with this module.

They seem to work fine together if I make sure to select a token before casting (which I should probably do anyway most of the time) and it's the other modules job to check that token exists before it tries to read properties from it regardless.

Thanks for getting the issue fixed, sorry again if I wasted any of your time looking for those console errors.
